mesonbuild/modules/cmake.py

```diff
@@ -272,6 +273,10 @@ def write_basic_package_version_file(self, state, _args, kwargs):
         if compatibility not in COMPATIBILITIES:
             raise mesonlib.MesonException('compatibility must be either AnyNewerVersion, SameMajorVersion or ExactVersion.')
 
+        arch_independent = kwargs.get('arch_independent', False)
+        if not isinstance(arch_independent, bool):
+            raise mesonlib.MesonException('arch_independent is not a bool.')
```

Comment:
Ideally we'd also revamp the function to use `@typed_kwargs` instead of `@permittedKwargs`, and then each kwarg would be a KwargInfo instead of a string, and the `KwargInfo` would contain attributes such as the allowed type (bool) and the default (False) and *also* the available since (0.62.0).

However, that is not your job (unless you want to do it for bonus points).

Reply:
I'll take this as a personal challenge :)

I'll either get a bit more familiar with the codebase or come back here crying for help